File: Chapter03/3.4 Refactor OrderProcessor/BadWay/OrderProcessor.cs
namespace Refactor.BadWay
{
    // The "monolithic" class that does everything
    public class OrderProcessor
    {
        public string Process(Order order)
        {
            // 1. Validation
            Console.WriteLine("(BadWay) Validating order...");
            if (order.Items.Count == 0 || order.Total <= 0)
            {
                throw new InvalidOperationException("Order is invalid.");
            }

            // 2. Payment Processing
            Console.WriteLine($"(BadWay) Processing payment for ${order.Total}...");
            bool paymentSuccess = true; // Simulating success

            // 3. Inventory Update & 4. Confirmation Email
            if (paymentSuccess)
            {
                Console.WriteLine("(BadWay) Updating inventory...");
                Console.WriteLine($"(BadWay) Sending confirmation email to {order.CustomerEmail}...");
                return "Order processed successfully.";
            }
            else
            {
                return "Payment failed.";
            }
        }
    }
}

File: Chapter 03/JS/3.4 OrderProcessor Refactor/Before/orderProcessor.js
class OrderProcessor {
    process(order) {
        console.log("  [Validate] Validating order...");
        if (order.items.length === 0 || order.total <= 0) {
            throw new Error("Order is invalid.");
        }

        console.log(`  [Payment] Processing payment for $${order.total.toFixed(2)}...`);
        const paymentSuccess = true;

        if (paymentSuccess) {
            console.log("  [Inventory] Updating inventory...");
            console.log(`  [Notify] Sending confirmation email to ${order.customerEmail}...`);
            return "Order processed successfully.";
        } else {
            return "Payment failed.";
        }
    }
}
module.exports = OrderProcessor;